Source/Dafny/Resolver.cs

```diff
@@ -18599,15 +18599,19 @@ public class SubsetConstraintGhostChecker : ProgramTraverser {
             }
 
             if (!constraintIsCompilable) {
```

Comment:
Not for this PR, but it would be nice if we can split `Resolver.cs` into multiple smaller segments, around the 3k line range, if only to let the syntax highlighting in GH work. Doing this using partial classes would be fine.

Reply:
Agreed. We could split it by phases.

Reply:
Yes, a split of `Resolver.cs` into several files, especially along the lines of phases (and sub-parts thereof), would be great!